lemur/plugins/lemur_openssh/plugin.py

```diff
@@ -0,0 +1,148 @@
+
+from lemur.utils import mktempfile
+from lemur.plugins import lemur_openssh as openssh
+from lemur.common.utils import parse_private_key , parse_certificate
+from lemur.plugins.lemur_cryptography.plugin import CryptographyIssuerPlugin
+from lemur.certificates.service import get_by_root_authority
+
+
+def run_process(command):
+    """
+    Runs a given command with pOpen and wraps some
+    error handling around it.
+    :param command:
+    :return:
+    """
+    p = subprocess.Popen(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    current_app.logger.debug(" ".join(command))
+    stdout, stderr = p.communicate()
+
+    if p.returncode != 0:
+        current_app.logger.error(stderr.decode())
+        raise Exception(stderr.decode())
+
+
+def split_cert(body):
+    """
+    To display certificate in Lemur website, we have to split
+    certificate in several line
+    :param body: certificate
+    :retur: splitted certificate
+    """
+    length = 65
+    return '\n'.join([body[i:i+length] for i in range(0, len(body), length)])
+
+
+def sign_certificate(common_name, public_key, authority_private_key, user, extensions, not_before, not_after):
+    with mktempfile() as issuer_tmp:
+        cmd = ['ssh-keygen', '-s', issuer_tmp]
+        with open(issuer_tmp, 'w') as i:
+            i.writelines(authority_private_key)
+        if 'extendedKeyUsage' in extensions and extensions['extendedKeyUsage'].get('useClientAuthentication'):
+            valid_interval = current_app.config.get("OPENSSH_VALID_INTERVAL_CLIENT", 1)  # 1 day by default
+            cmd.extend(['-I', user['username'] + ' user key',
+                        '-n', user['username']])
+        else:
+            valid_interval = current_app.config.get("OPENSSH_VALID_INTERVAL_SERVER", 14)  # 2 weeks by default
+            domains = {common_name}
+            for name in extensions['subAltNames']['names']:
+                if name['nameType'] == 'DNSName':
+                    domains.add(name['value'])
+            cmd.extend(['-I', common_name + ' host key',
+                        '-n', ','.join(domains),
+                        '-h'])
+        # something like 20201024
+        ssh_not_before = datetime.fromisoformat(not_before).strftime("%Y%m%d")
+        cert_not_after = datetime.fromisoformat(not_after).strftime("%Y%m%d")
+        ssh_not_after = (datetime.now() + timedelta(days=valid_interval)).strftime("%Y%m%d")
+        ssh_not_after = min(ssh_not_after, cert_not_after)
+        cmd.extend(['-V', ssh_not_before + ':' + ssh_not_after])
+        with mktempfile() as cert_tmp:
+            with open(cert_tmp, 'w') as f:
+                f.write(public_key)
+
+            cmd.append(cert_tmp)
+            run_process(cmd)
+    pub = cert_tmp + '-cert.pub'
+    with open(pub, 'r') as p:
+        body = split_cert(p.read())
+    unlink(pub)
+    return body
+
+
+class OpenSSHIssuerPlugin(CryptographyIssuerPlugin):
+    title = "OpenSSH"
+    slug = "openssh-issuer"
```

Comment:
isn't this missing `create_certificate()`?

Reply:
No because certificate is a common x509 created by CryptographyIssuerPlugin classes.

I fact authorities are also created by CryptographyIssuerPlugin, I just check if user doesn't try to create subca authority.